kroxylicious-integration-tests/src/test/java/io/kroxylicious/proxy/FilterIT.java

```diff
@@ -278,6 +280,74 @@ void supportsForwardDeferredByAsynchronousBrokerRequest(RequestResponseMarkingFi
+        var markingFilter = new FilterDefinitionBuilder(RequestResponseMarkingFilterFactory.class.getName())
+                .withConfig("keysToMark", Set.of(LIST_TRANSACTIONS),
+                        "direction", Set.of(direction),
+                        "name", name,
+                        "forwardingStyle", forwardingStyle)
+                .build();
+        try (var tester = mockKafkaKroxyliciousTester((mockBootstrap) -> proxy(mockBootstrap).addToFilters(markingFilter));
+                var kafkaClient = tester.simpleTestClient()) {
+            tester.addMockResponseForApiKey(new ResponsePayload(LIST_TRANSACTIONS, LIST_TRANSACTIONS.latestVersion(), new ListTransactionsResponseData()));
+            ApiVersionsResponseData apiVersionsResponseData = new ApiVersionsResponseData();
+            apiVersionsResponseData.apiKeys()
+                    .add(new ApiVersionsResponseData.ApiVersion().setApiKey(FETCH.id).setMaxVersion(FETCH.latestVersion()).setMinVersion(FETCH.oldestVersion()));
+            tester.addMockResponseForApiKey(new ResponsePayload(API_VERSIONS, API_VERSIONS.latestVersion(), apiVersionsResponseData));
+
+            // In the ASYNCHRONOUS_REQUEST_TO_BROKER case, the filter will send an async list_group
+            // request to the broker and defer the forward of the list transaction response until the list groups
+            // response arrives.
+            if (forwardingStyle == ForwardingStyle.ASYNCHRONOUS_REQUEST_TO_BROKER) {
+                tester.addMockResponseForApiKey(new ResponsePayload(LIST_GROUPS, LIST_GROUPS.latestVersion(), new ListGroupsResponseData()));
+            }
+
+            var responseAFuture = kafkaClient
+                    .get(new Request(LIST_TRANSACTIONS, LIST_TRANSACTIONS.latestVersion(), "client", new ListTransactionsRequestData()));
+
+            var responseBFuture = kafkaClient
+                    .get(new Request(LIST_TRANSACTIONS, LIST_TRANSACTIONS.latestVersion(), "client", new ListTransactionsRequestData()));
+            Response responseA = responseAFuture.get(5, TimeUnit.SECONDS);
+            Response responseB = responseBFuture.get(5, TimeUnit.SECONDS);
+
+            List<Request> listTransactionRequests = tester.getRequestsForApiKey(LIST_TRANSACTIONS);
+            var requestMessageReceivedByBrokerA = listTransactionRequests.get(0).message();
+            String dispatchThreadA = getDispatchThreadName(direction, responseA, requestMessageReceivedByBrokerA);
+            var requestMessageReceivedByBrokerB = listTransactionRequests.get(1).message();
+            String dispatchThreadB = getDispatchThreadName(direction, responseB, requestMessageReceivedByBrokerB);
+            assertThat(dispatchThreadA).isNotEmpty().containsIgnoringCase("eventloop");
+            assertThat(dispatchThreadB).isNotEmpty().containsIgnoringCase("eventloop")
+                    .describedAs("filter invocations should be dispatched from the same thread").isEqualTo(dispatchThreadA);
```

Comment:
What do you think about checking the thread name contains eventloop? It could break if netty changes their thread naming.

From within the Filter you can't really tell if the dispatching thread is the thread associated with the underlying channel. All we can do is check that the same thread is dispatching all the methods.

Reply:
The test filter could execute a task on `io.kroxylicious.proxy.filter.FilterFactoryContext#eventLoop` to obtain the event loop thread.  That would mean your DISPATCH_THREAD tag could turned into ON_EVENT_LOOP true|false.  That would avoid the knowledge of the event loop thread name.

Separately, I've been mulling if FilterFactoryContext#inEventLoop() would make a useful addition to the API.  It would allow a filter author to make their own threading decisions (i.e. when to delegate to #eventLoop).  If it were to exist, it would also give you an easy way to observe if you were indeed running on the event loop.

Reply:
> FilterFactoryContext#inEventLoop() would make a useful addition to the API.

Tom was nitpicking the netty-ish eventloop language too in another PR, in the module javadoc we say the filter instance is associated with a thread that does all the dispatch/construction, maybe we should find a new name like `dispatchExecutor` or `connectionThreadExecutor`  that ties into that description better and dodges the netty naming? Might also be nice to include them in the `FilterContext` as well as `FilterFactoryContext` in future.

Reply:
> The test filter could execute a task on io.kroxylicious.proxy.filter.FilterFactoryContext#eventLoop to obtain the event loop thread. 

I found this a bit of a problem, I tried to run a task during construction to obtain the thread, but we're in the eventloop itself so we can't wait on the result. It would have to be done async and might race with the first request. You could change the onRequest/onResponse methods to record the current thread and execute the rest on the work using the eventloop then check you haven't changed context, but then I think we are starting to interfere with what the RequestResponseMarkingFilter is trying to do, controlling precisely when we change threads. Maybe we could write another Filter but the RequestResponseMarkingFilter is handy for checking all the types of forwarding execute on the eventloop.

If we added support for inEventLoop() it would be straightforward, for now I think checking the thread names is good enough.

Reply:
Ok